File: Neha Niharika Kar/Python Code/Atm Pressure Monitor.py
from environment import *
from physical import *
from gpio import *
from time import *
from ioeclient import *
import math

ENVIRONMENT_NAME = "Atmospheric Pressure"
P_MIN = 0;    # assumed minimum pressure value in the environment
P_MAX = 100*1000;  # (Pa) assumed maximum pressure value in the environment
level = 0

def setup():

    IoEClient.setup({
        "type": "Atm. Pressure Sensor",
        "states": [
        {
            "name": "Atm. Pressure",
            "type": "number",
            "controllable": False
        }]
    })

    sendReport()



def main():
    setup()
    while True:
        detect()
        sendReport()
        delay(1000)


def detect():
    global ENVIRONMENT_NAME
    value = Environment.get(ENVIRONMENT_NAME) * 1000; # sale to Pa from kPa
    if value >= 0 :
        setLevel( value )

def sendReport():
    global level
    report = level; # comma seperated states

    IoEClient.reportStates(report)
    setDeviceProperty(getName(), "level", str(level) + " Pa")


def setLevel(newLevel):
    global level
    global ENVIRONMENT_NAME
    global P_MIN
    global P_MAX
    if  level == newLevel:
        return

    level = math.floor(float(newLevel))

    avalue = level

    if avalue < P_MIN:
        avalue = P_MIN
    if avalue > P_MAX:
        avalue = P_MAX

    setCustomText(20,18,200,20, Environment.getValueWithUnit(ENVIRONMENT_NAME) )
    # 50% at sea level
    out_avalue = float(128)/101.325 * Environment.getMetricValue(ENVIRONMENT_NAME)

    analogWrite(A0, out_avalue)
    sendReport()
    delay(500)

if __name__ == "__main__":
	main()

File: Neha Niharika Kar/JavaScript Code/Atm Pressure Monitor.js
var ENVIRONMENT_NAME = "Atmospheric Pressure";
var P_MIN = 0;    // assumed minimum pressure value in the environment
var P_MAX = 100*1000;  // (Pa) assumed maximum pressure value in the environment

var level = 0;
var unit;

function setup() {
	
	IoEClient.setup({
		type: "Atm. Pressure Sensor",
		states: [
		{
			name: "Atm. Pressure",
			type: "number",
			unit: unit,
			controllable: false
		}]
	});
	

//	setCustomText(30,18,200,20, level );
	
	sendReport();

}

function loop() {

	detect();
	sendReport();
	delay(1000);
}

function detect()
{
	var value = Environment.get(ENVIRONMENT_NAME) * 1000; // sale to Pa from kPa
	if (value >= 0 )
		setLevel( value );
}

function sendReport()
{
	var report = level;	// comma seperated states
	
	IoEClient.reportStates(report);
	setDeviceProperty(getName(), "level", level + " Pa");
}

function setLevel(newLevel)
{
	if  (level == newLevel)
		return;
		
	level = Math.floor(parseFloat(newLevel));
	
	var avalue = level;
	
	if(avalue < P_MIN)
		avalue = P_MIN;
	if(avalue > P_MAX)
		avalue = P_MAX;
		
	setCustomText(20,18,200,20, Environment.getValueWithUnit(ENVIRONMENT_NAME) );
	
	
//	var out_avalue = Math.floor((avalue - P_MIN)/(P_MAX - P_MIN) * 255);
	// 50% at sea level
	var out_avalue = 128/101.325 * Environment.getMetricValue(ENVIRONMENT_NAME);
	
	analogWrite(A0, out_avalue);

//	Serial.println(Environment.getValueWithUnit(ENVIRONMENT_NAME) +", "+out_avalue);
	
	sendReport();
	delay(500);
}

